app/logic/report_logic.rb

```diff
@@ -163,13 +163,32 @@ def report_software_versions
       
       batch_created_at = Batch.find_by(uuid: p.payload[:batch_uuid])&.created_at
 
-      sql = "SELECT software_version, count(*) as count, SUM(active_stake) as as_sum
-             FROM validator_score_v1s 
-             WHERE network = '#{p.payload[:network]}'
-             AND updated_at > '#{batch_created_at}'
-             GROUP BY software_version;"
-      # Grab the vote history for this batch
-      software_versions_with_active_stake = ValidatorScoreV1.connection.execute(sql)
+      # Get validator ids for current batch.
+      validator_ids_for_batch_sql = %Q{
+        SELECT v.id
+        FROM vote_account_histories as vah
+        INNER JOIN vote_accounts as va
+        ON vah.vote_account_id = va.id
+        INNER JOIN validators as v
+        ON va.validator_id = v.id
+        WHERE vah.network = '#{p.payload[:network]}'
```

Comment:
perhaps you should use variable instead of direct SQL injection like: WHERE vah.network = :network